File: src/main/java/com/heekng/celloct/dto/ShopDto.java
package com.heekng.celloct.dto;

import com.heekng.celloct.entity.Shop;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

public class ShopDto {

    @Getter
    @Setter
    @NoArgsConstructor
    public static class CreateRequest {
        private String name;
        private String phone;
        private String info;
        private String managerName;

        @Builder
        public CreateRequest(String name, String phone, String info, String managerName) {
            this.name = name;
            this.phone = phone;
            this.info = info;
            this.managerName = managerName;
        }

        public Shop toEntity() {
            return Shop.Companion.fixture(phone, name, info);
        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class UpdateRequest {
        private Long id;
        private String phone;
        private String info;

        @Builder
        public UpdateRequest(Long id, String phone, String info) {
            this.id = id;
            this.phone = phone;
            this.info = info;
        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class ListResponse {
        private Long id;
        private String name;
        private String info;
        private Integer staffCount;

        @Builder
        public ListResponse(Shop shop) {
            this.id = shop.getId();
            this.name = shop.getName();
            this.info = shop.getInfo();
            this.staffCount = shop.getStaffList().size();
        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class ShopDetailResponse {
        private Long id;
        private String phone;
        private String name;
        private String info;

        @Builder
        public ShopDetailResponse(Shop shop) {
            this.id = shop.getId();
            this.phone = shop.getPhone();
            this.name = shop.getName();
            this.info = shop.getInfo();
        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class HomeShopManagerResponse {
        private Long id;
        private String name;

        @Builder
        public HomeShopManagerResponse(Shop shop) {
            this.id = shop.getId();
            this.name = shop.getName();
        }

    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class HomeShopStaffResponse {
        private Long id;
        private String name;

        @Builder
        public HomeShopStaffResponse(Shop shop) {
            this.id = shop.getId();
            this.name = shop.getName();
        }

    }

}

File: src/main/kotlin/com/heekng/celloct/dto/ShopDto.kt
package com.heekng.celloct.dto

import com.heekng.celloct.entity.Shop
import com.heekng.celloct.entity.Shop.Companion.fixture

class ShopDto {
    data class CreateRequest (
        val name: String? = null,
        val phone: String? = null,
        val info: String? = null,
        val managerName: String? = null,
    ) {
        fun toEntity(): Shop {
            return fixture(
                phone = phone,
                name = name ?: "name",
                info = info
            )
        }
    }

    data class UpdateRequest (var id: Long?, val phone: String, val info: String)

    data class ListResponse (
        val id: Long?,
        val name: String,
        val info: String?,
        val staffCount: Int,
    ) {
        constructor(shop: Shop): this(
            id = shop.id,
            name = shop.name,
            info = shop.info,
            staffCount = shop.staffList.size,
        )
    }

    data class ShopDetailResponse (
        val id: Long?,
        val phone: String?,
        val name: String,
        val info: String?,
    ) {
        constructor(shop: Shop): this(
            id = shop.id,
            phone = shop.phone,
            name = shop.name,
            info = shop.info,
        )
    }

    data class HomeShopManagerResponse (
        val id: Long?,
        val name: String,
    ) {
        constructor(shop: Shop): this(
            id = shop.id,
            name = shop.name,
        )
    }

    data class HomeShopStaffResponse (
        val id: Long?,
        val name: String,
    ) {
        constructor(shop: Shop): this(
            id = shop.id,
            name = shop.name,
        )
    }
}